DigitalLearningSolutions.Data.Tests/Services/DiagnosticAssessmentDataServiceTests.cs

```diff
@@ -316,6 +326,325 @@ public void Get_diagnostic_assessment_should_get_tutorialIds_ordered_by_orderBy(
+                // Given
+                const int candidateId = 1;
+                const int customisationId = 8194;
+                const int sectionId = 216;
+
+                tutorialContentTestHelper.RemoveCustomisationTutorial(customisationId, 923);
+                tutorialContentTestHelper.RemoveCustomisationTutorial(customisationId, 924);
+                tutorialContentTestHelper.RemoveCustomisationTutorial(customisationId, 925);
+                tutorialContentTestHelper.RemoveCustomisationTutorial(customisationId, 926);
+                tutorialContentTestHelper.RemoveCustomisationTutorial(customisationId, 927);
+                tutorialContentTestHelper.RemoveCustomisationTutorial(customisationId, 928);
+
+                // All in section 216
+                // Tutorial: 927  OrderByNumber 34
+                // Tutorial: 928  OrderByNumber 35
+                // Tutorial: 929  OrderByNumber 35
+                // Tutorial: 930  OrderByNumber 36
+                const int nextTutorialId = 929;
+
+                // When
+                var result = diagnosticAssessmentDataService.GetDiagnosticAssessment(customisationId, candidateId, sectionId);
+
+                // Then
+                result.Should().NotBeNull();
+                result!.NextTutorialId.Should().Be(nextTutorialId);
+            }
+        }
+
+        [Test]
+        public void Get_diagnostic_assessment_nextTutorial_returns_tutorialId_after_shared_orderByNumber()
+        {
+            using (new TransactionScope())
+            {
+                // Given
+                const int candidateId = 1;
+                const int customisationId = 8194;
+                const int sectionId = 216;
+
+                tutorialContentTestHelper.RemoveCustomisationTutorial(customisationId, 923);
+                tutorialContentTestHelper.RemoveCustomisationTutorial(customisationId, 924);
+                tutorialContentTestHelper.RemoveCustomisationTutorial(customisationId, 925);
+                tutorialContentTestHelper.RemoveCustomisationTutorial(customisationId, 926);
+                tutorialContentTestHelper.RemoveCustomisationTutorial(customisationId, 927);
+                tutorialContentTestHelper.RemoveCustomisationTutorial(customisationId, 928);
+                tutorialContentTestHelper.RemoveCustomisationTutorial(customisationId, 929);
+
+                // All in section 216
+                // Tutorial: 927  OrderByNumber 34
+                // Tutorial: 928  OrderByNumber 35
+                // Tutorial: 929  OrderByNumber 35
+                // Tutorial: 930  OrderByNumber 36
+                const int nextTutorialId = 930;
+
+                // When
+                var result = diagnosticAssessmentDataService.GetDiagnosticAssessment(customisationId, candidateId, sectionId);
+
+                // Then
+                result.Should().NotBeNull();
+                result!.NextTutorialId.Should().Be(nextTutorialId);
+            }
+        }
+
```

Comment:
This file is getting quite long, might be worth splitting it up, e.g. by moving all the tests for next tutorial/section into their own test file. You can use a partial class so you can still share the setup, like the tests for the learning portal controller 🙂